src/store/movie/movieActions.js

```diff
@@ -0,0 +1,178 @@
+            })
+            .catch(error => {
+                dispatch(movieRequestFailure(error.message))
+            })
+    }
+}
+
+export const deleteMovie = (id) => {
+    return dispatch => {
+        axios
+            .delete(`${MOVIE_BASE_URL}/${id}`)
+            .then(()=>{
+                dispatch(fetchMovies())
+            })
+            .catch(error => {
+                dispatch(movieRequestFailure(error.message))
+            })
+    }
+}
+
+export const updateMovie = (movie) => {
+    return dispatch => {
+        if (movie.tagline == null || movie.tagline === '') {
+            movie.tagline = "default"
+        }
+        axios
+            .put(MOVIE_BASE_URL, movie)
+            .then((response) => {
+                dispatch(fetchMovies())
+            })
+            .catch(error => {
+                dispatch(movieRequestFailure(error))
+            })
+    }
+}
+
+export const createMovie = (movie) => {
+    return dispatch => {
+        axios
+            .post(MOVIE_BASE_URL, movie)
+            .then(()=>{
+                dispatch(fetchMovies())
+            })
+            .catch(error => {
+                dispatch(movieRequestFailure(error))
+            })
+    }
+}
+
+export const setSortParam = sortParams => {
+    return {
+        type: SET_SORT_PARAMS,
+        payload: sortParams
+    }
+}
+
+export const fetchSort = sortText => {
+    return dispatch => {
+        const updateWithSort = sortParams => {
+            dispatch(setSortParam(sortParams))
+                dispatch(fetchMovies())
+        }
+        switch (sortText) {
+            case 'NAME':
+                updateWithSort(SORT_TITLE_PARAM)
+                return
+            case 'RELEASE DATE':
+                updateWithSort(SORT_DATE_PARAM)
+                return
+            case 'RATING':
+                updateWithSort(SORT_RATING_PARAM)
+                return
+            default:
+                updateWithSort(SORT_LATEST_PARAM)
+                return
+        }
+    }
+}
+
+export const setFilterParam = filterParam => {
+    return {
+        type: SET_FILTER_PARAM,
+        payload: filterParam
+    }
+}
+
+export const fetchFilter = filterText => {
+    return dispatch => {
+        const setFilterAndFetch = filter => {
+            dispatch(setFilterParam(filter))
+            dispatch(fetchMovies())
+
+        }
+        switch(filterText){
+            case 'DOCUMENTARY':
+                setFilterAndFetch(FILTER_DOCUMENTARY_PARAM)
+                return
+            case 'COMEDY':
+                setFilterAndFetch(FILTER_COMEDY_PARAM)
+                return
+            case 'CRIME':
+                setFilterAndFetch(FILTER_CRIME_PARAM)
+                return
+            case 'HORROR':
+                setFilterAndFetch(FILTER_HORROR_PARAM)
+                return
+            default:
+                setFilterAndFetch(FILTER_NONE)
+                return        
+        }
+        
+    }
+}
```

Comment:
Present a good practice to split actions and reducers into different files.

Reply:
do not get it, reducer is in separate file - movieReducer.js and all actions are in movieActions.js